File: hw2/task4/java/Task4Monster.java
public class Task4Monster extends Monster {
    public Task4Monster(int monsterID, int healthCapacity) {
        super(monsterID, healthCapacity);
    }

    @Override
    public void dropItems(Soldier soldier) {
        super.dropItems(soldier);
        ((Task4Soldier) soldier).addCoin();
    }
}

File: hw2/task4/python/Task4Monster.py
"""
 * CSCI3180 Principles of Programming Languages
 *
 * --- Declaration ---
 *
 * I declare that the assignment here submitted is original except for source
 * material explicitly acknowledged. I also acknowledge that I am aware of
 * University policy and regulations on honesty in academic work, and of the
 * disciplinary guidelines and procedures applicable to breaches of such policy
 * and regulations, as contained in the website
 * http://www.cuhk.edu.hk/policy/academichonesty/
 *
 * Assignment 2
 * Name : Chao Yu
 * Student ID : 1155053722
 * Email Addr : ychao5@cse.cuhk.edu.hk
"""

from Monster import Monster

class Task4Monster(Monster):
    def __init__(self, monster_id, health_capacity):
        super().__init__(monster_id, health_capacity)

    # Override.
    def drop_items(self, soldier):
        super().drop_items(soldier)
        soldier.add_coin()
